src/test/javascript/spec/component/complaints/complaints-form.component.spec.ts

```diff
@@ -0,0 +1,104 @@
+import { async, ComponentFixture, fakeAsync, TestBed, tick } from '@angular/core/testing';
+import { ComplaintService, EntityResponseType } from 'app/complaints/complaint.service';
+import { MockComplaintService } from '../../helpers/mocks/service/mock-complaint.service';
+
+import { TranslateModule } from '@ngx-translate/core';
+import { ComplaintsFormComponent } from 'app/complaints/form/complaints-form.component';
+import { ArtemisTestModule } from '../../test.module';
+import { ArtemisSharedModule } from 'app/shared/shared.module';
+import { Exercise } from 'app/entities/exercise.model';
+import { Course } from 'app/entities/course.model';
+import { of, throwError } from 'rxjs';
+import { HttpErrorResponse } from '@angular/common/http';
+import { MockProvider } from 'ng-mocks';
+import { AlertService } from 'app/core/util/alert.service';
+
+describe('ComplaintsComponent', () => {
+    const teamComplaints = 42;
+    const studentComplaints = 69;
+    const course: Course = { maxTeamComplaints: teamComplaints, maxComplaints: studentComplaints };
+    const exercise: Exercise = { id: 1, teamMode: false } as Exercise;
+    const courseExercise: Exercise = { id: 1, teamMode: false, course } as Exercise;
+    const courseTeamExercise: Exercise = { id: 1, teamMode: true, course } as Exercise;
+    let component: ComplaintsFormComponent;
+    let fixture: ComponentFixture<ComplaintsFormComponent>;
+    let complaintService: ComplaintService;
+    let alertService: AlertService;
+
+    beforeEach(async(() => {
+        TestBed.configureTestingModule({
+            imports: [TranslateModule.forRoot(), ArtemisTestModule, ArtemisSharedModule],
```

Comment:
i do not think `TranslateModule.forRoot()` is neede as in imports. you should not import huge production modules directly like `ArtemisSharedModule`. please just mock the classess you need.

Reply:
Thanks, I used already existing test classes and missed this import. 👍